File: SampleCode/VisaAPICalls/Node/VisaCheckoutSample.js
var request = require('request');

var callId = 'put your call id here';
var apiKey = 'put your api key here';
var sharedSecret = 'put your shared secret here';
var baseUri = 'wallet-services-web/';
var resourcePath = 'payment/data/' + callId;
var queryParams = 'apikey=' + apiKey;

var timestamp = Math.floor(Date.now() / 1000);
var preHashString = sharedSecret + timestamp + resourcePath + queryParams;
var crypto = require('crypto');
var hashString = crypto.createHash('sha256').update(preHashString).digest('hex');
var xPayToken = 'x:' + timestamp + ':' + hashString;

var url = 'https://sandbox.api.visa.com/' + baseUri + resourcePath + '?' + queryParams;
var headers = {'content-type': 'application/json', 'accept':'application/json','x-pay-token': xPayToken} 
var options = {
  url: url,
  method: "GET",
  headers: headers,
  json: true
};
 
function callback(error, response, body) {
  if (!error) {
    console.log(response.statusCode);
    console.log(response.statusMessage);
    console.log(body);
  } else {
    console.log("Got error: " + error);
  }
}

request(options, callback);

File: SampleCode/VisaAPICalls/Python/VisaCheckoutSample.py
import requests
import calendar
import time
import datetime
from hashlib import sha256

def generate_timestamp():
    d = datetime.datetime.utcnow()
    timestamp = calendar.timegm(d.timetuple())
    return str(timestamp)

def generate_x_pay_token(shared_secret, resource_path, query_params, body):
    timestamp = generate_timestamp()
    pre_hash_string = shared_secret + timestamp + resource_path + query_params + body
    hash_string = sha256(pre_hash_string.rstrip()).hexdigest()
    return 'x:' + timestamp + ':' + hash_string

def get_payment_info(shared_secret, api_key, call_id):
    base_uri = 'wallet-services-web/'
    resource_path = 'payment/data/' + call_id
    data_level = 'FULL'
    query_params = 'apikey=' + api_key + '&dataLevel=' + data_level
    body = ''
    x_pay_token = generate_x_pay_token(shared_secret, resource_path, query_params, body)	
    headers = {'content-type': 'application/json', 'accept':'application/json', 'x-pay-token': x_pay_token}
    r = requests.get('https://sandbox.api.visa.com/' + base_uri + resource_path + '?' + query_params, headers = headers)
    print r.status_code
    print r.content

def main():
    call_id = 'put your call id here'
    api_key = 'put your api key here'
    shared_secret = 'put your shared secret here'	
    get_payment_info(shared_secret, api_key, call_id)

if __name__ == '__main__':
    main()